api/project.go

```diff
@@ -137,6 +139,7 @@ type ProjectPatch struct {
 	Name         *string              `jsonapi:"attr,name"`
 	Key          *string              `jsonapi:"attr,key"`
 	WorkflowType *ProjectWorkflowType `jsonapi:"attr,workflowType"`
+	RoleProvider *string              `jsonapi:"attr,roleProvider"`
```

Comment:
Setting to *string is to avoid a jsonapi's Bug. Same with the field `RowStatus`
`jsonapi: Can't unmarshal GITLAB_SELF_HOST (string) to struct field `RoleProvider`, which is a pointer to `ProjectRoleProvider (string)`

Reply:
I see. I'm curious why "WorkflowType *ProjectWorkflowType" would work above. You can make a comment or link to the related bug at least.

Reply:
It works for `WorkflowType *ProjectWorkflowType` since we never try to unmarshal this field.